applications/notifications/services.py

```diff
@@ -0,0 +1,43 @@
+from django.db import transaction
+import environ
+from onesignal_sdk.client import Client as OneSignalClient
+
+from .models import Notification
+
+env = environ.Env()
+
+
+class NotificationSender(object):
+    client = OneSignalClient(app_id=env('ONESIGNAL_APP_ID'), rest_api_key=env('ONESIGNAL_API_KEY'))
+
+    def send_notification(self, emails, data, message):
+        body = self._body(emails, data, message)
+        self.client.send_notification(body)
+
+    def _body(self, emails, data, message):
+        return {
+            'filters': self._create_filter_by_emails(emails),
+            'data': data,
+            'contents': {'en': message},
+        }
+
+    def _create_filter_by_emails(self, emails):
+        email_list = emails if type(emails) is list else [emails]
+
+        def email_filter(email):
+            return {'field': 'tag', 'key': 'email', 'relation': '=', 'value': email}
+
+        def filter_option(i):
+            return email_filter(email_list[int(i/2)]) if i % 2 == 0 else {'operator': 'OR'}
+
+        return [filter_option(i) for i in range(len(email_list*2) - 1)]
```

Comment:
Sorry but I don't follow what this does. Why are you doubling the length the list or checking if the index is even?

Suggestion, maybe not aplicable here, when wanting to use the indexes of arrays use `for i, x in enumerate(list)` instead of range. That will return the index and the element of the list.

Reply:
I update the PR description adding an example about what this method tries to do. 
I doubled the list length because I have to create the `filter` value, so, when the index is even I add the email, and when not I add the `OR` value.